### PR TITLE
fix search bar, add preconnect functionality + linting

### PR DESCRIPTION
**Pull request by sarracini**

### CODE REVIEW

File: browser/components/newtab/RemoteNewTabLocation.jsm [hunk 1]
```diff
@@ -7,7 +7,7 @@ this.EXPORTED_SYMBOLS = ["RemoteNewTabLocation"];
 Components.utils.import("resource://gre/modules/Services.jsm");
 Components.utils.importGlobalProperties(["URL"]);
 
-const DEFAULT_PAGE_LOCATION = "https://mozilla.github.io/remote-newtab/";
+const DEFAULT_PAGE_LOCATION = "http://localhost:8000";
```
Comment (marcoscaceres) [inline browser/components/newtab/RemoteNewTabLocation.jsm]: Note to self: Need to revert this before merging.

File: browser/modules/SearchProvider.jsm [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        });
+      }
+      return state;
+    }.bind(this));
+
+    return t;
+  },
+
+  performSearch: function(browser, data) {
+    this._ensureDataHasProperties(data, [
```
Comment (marcoscaceres) [inline browser/modules/SearchProvider.jsm]: Gong to drop this. This is alien. Errors should occur if the properties are missing in place.

File: browser/modules/SearchProvider.jsm [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+const MAX_LOCAL_SUGGESTIONS = 3;
+const MAX_SUGGESTIONS = 6;
+// splits data urls into component parts [url, type, data]
+const dataURLParts = /(?:^data:)(.+)(?:;base64,)(.*)$/;
+const l10nKeysNames = [
+  "searchHeader",
+  "searchPlaceholder",
+  "searchWithHeader",
+  "searchSettings",
+  "searchForSomethingWith"
```
Comment (marcoscaceres) [inline browser/modules/SearchProvider.jsm]: missing ","

File: browser/modules/SearchProvider.jsm [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+  _suggestionMap: new WeakMap(),
+
+  _searchSuggestionUIStrings: new Map(),
+
+  /**
+   * Makes a copy of the current search suggestions.
+   * @return {Object} Key/value pairs representing the search suggestions.
+   */
+  get searchSuggestionUIStrings() {
+    var result = Object.create(null);
```
Comment (marcoscaceres) [inline browser/modules/SearchProvider.jsm]: let

**Comment by marcoscaceres** [comment]
R+ WOOT WOOT!!

**Comment by marcoscaceres** [comment]
@oyiptong I think this go trashed too when master was `-f` updated... I think I'm going to put branch protection on again :wink:

**Comment by oyiptong** [comment]
Hahaha

**Pull request merged**